debezium-core/src/main/java/io/debezium/pipeline/signal/channels/KafkaSignalChannel.java

```diff
@@ -163,15 +172,20 @@ public void reset(Object reference) {
 
     private static Configuration buildKafkaConfiguration(Configuration signalConfig) {
 
-        return signalConfig.subset(CONSUMER_PREFIX, true).edit()
+        Configuration.Builder confBuilder = signalConfig.subset(CONSUMER_PREFIX, true).edit()
                 .withDefault(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, signalConfig.getString(BOOTSTRAP_SERVERS))
                 .withDefault(ConsumerConfig.CLIENT_ID_CONFIG, UUID.randomUUID().toString())
                 .withDefault(ConsumerConfig.GROUP_ID_CONFIG, signalConfig.getString(GROUP_ID))
                 .withDefault(ConsumerConfig.FETCH_MIN_BYTES_CONFIG, 1) // get even the smallest message
                 .withDefault(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false)
                 .withDefault(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 10000) // readjusted since 0.10.1.0
                 .withDefault(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class)
-                .withDefault(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class)
+                .withDefault(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        if (signalConfig.getBoolean(SIGNAL_CONSUMER_OFFSET_COMMIT_ENABLED)) {
+            confBuilder.withDefault(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, true)
+                    .withDefault(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
```

Comment:
Is this enough? Should this act only when there is no initial offset in Kafka or if an offset is out of range? What about the restart after some processed records? Should be an explicit commit or auto commit present?

Reply:
Sorry, just saw the auto commit setting. Don't mind mu comment.